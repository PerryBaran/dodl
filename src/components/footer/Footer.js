import React, { useContext } from 'react';
import style from './footer.module.css'
import AppContext from '../../utils/context/AppContext';

const Footer = (props) => {
    const { hideWhilePlaying } = useContext(AppContext);

    return (
        <footer>
            <a className={`${style.attribution} ${style.left} ${hideWhilePlaying(style.pause)}`} 
                href='https://distrokid.com/hyperfollow/jenico/dreaming-of-detuned-love'
                target={'_blank'}
                rel="noreferrer">
                    Music by Jenico
            </a>
            <a className={`${style.attribution} ${style.right} ${hideWhilePlaying(style.pause)}`} 
                href='https://www.artstation.com/tommartyn'
                target={'_blank'}
                rel="noreferrer">
                    Art by Tom Martyn
            </a>
        </footer>
    );
}

export default Footer;