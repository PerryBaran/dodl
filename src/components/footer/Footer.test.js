import React from "react";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom'
import Footer from "./Footer";
import AppContext from "../../utils/context/AppContext";
import renderer from "react-test-renderer";

afterEach(cleanup);

const renderFooter = (input) => {
    const isPlaying = input | false
    return (
        <AppContext.Provider value={{isPlaying}}>
            <Footer/>
        </AppContext.Provider>
    )
}

it("renders without crashing", () => { 
    render(renderFooter())
})

it("matches snapshot, isPlaying className false", () => {
    const tree = renderer.create(renderFooter()).toJSON();
    expect(tree).toMatchSnapshot();
})

it("matches snapshot, isPlaying className true", () => {
    const tree = renderer.create(renderFooter(true)).toJSON();
    expect(tree).toMatchSnapshot();
})
