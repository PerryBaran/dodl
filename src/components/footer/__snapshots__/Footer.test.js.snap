// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, isPlaying className false 1`] = `
<footer>
  <a
    className="attribution left pause"
    href="https://distrokid.com/hyperfollow/jenico/dreaming-of-detuned-love"
    rel="noreferrer"
    target="_blank"
  >
    Music by Jenico
  </a>
  <a
    className="attribution right pause"
    href="https://www.artstation.com/tommartyn"
    rel="noreferrer"
    target="_blank"
  >
    Art by Tom Martyn
  </a>
</footer>
`;

exports[`matches snapshot, isPlaying className true 1`] = `
<footer>
  <a
    className="attribution left "
    href="https://distrokid.com/hyperfollow/jenico/dreaming-of-detuned-love"
    rel="noreferrer"
    target="_blank"
  >
    Music by Jenico
  </a>
  <a
    className="attribution right "
    href="https://www.artstation.com/tommartyn"
    rel="noreferrer"
    target="_blank"
  >
    Art by Tom Martyn
  </a>
</footer>
`;
